src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -219,29 +224,199 @@ export default class SourceFileLinter {
+				nodeToHighlight = apiVersionNode;
+			}
+
+			if (nodeToHighlight) {
+				// reporter.addMessage() won't work in this case as it's bound to the current analyzed file.
+				// The findings can be in different file i.e. Control being analyzed,
+				// but reporting might be in ControlRenderer
+				const nodeSourceFile = nodeToHighlight.getSourceFile();
+				const nodeSourceMap = this.#sourceMaps?.get(nodeSourceFile.fileName);
+				this.#context.addLintingMessage(
+					nodeSourceFile.fileName, MESSAGE.NO_DEPRECATED_RENDERER, undefined as never,
+					getPositionsForNode({
+						node: nodeToHighlight,
+						sourceFile: nodeSourceFile,
+						resourcePath: nodeSourceFile.fileName,
+						traceMap: nodeSourceMap ? new TraceMap(nodeSourceMap) : undefined,
+					}).start);
 			}
+		// Analyze renderer property when it's a function i.e. { renderer: () => {} }
+		} else if (ts.isMethodDeclaration(node) || ts.isArrowFunction(node) ||
+			ts.isFunctionExpression(node) || ts.isFunctionDeclaration(node)) {
+			// reporter.addMessage() won't work in this case as it's bound to the current analyzed file.
+			// The findings can be in different file i.e. Control being analyzed,
+			// but reporting might be in ControlRenderer
+			const nodeSourceFile = node.getSourceFile();
+			const nodeSourceMap = this.#sourceMaps?.get(nodeSourceFile.fileName);
+			this.#context.addLintingMessage(
+				nodeSourceFile.fileName, MESSAGE.NO_DEPRECATED_RENDERER, undefined as never,
+				getPositionsForNode({
+					node,
+					sourceFile: nodeSourceFile,
+					resourcePath: nodeSourceFile.fileName,
+					traceMap: nodeSourceMap ? new TraceMap(nodeSourceMap) : undefined,
+				}).start);
+		}
+
+		this.analyzeIconCallInRenderMethod(node);
+	}
 
-			if (initializerType.flags & ts.TypeFlags.StringLiteral) {
-				let rendererName;
-				if (
-					ts.isStringLiteral(rendererMember.initializer) ||
-					ts.isNoSubstitutionTemplateLiteral(rendererMember.initializer)
-				) {
-					rendererName = rendererMember.initializer.text;
+	// If there's an oRm.icon() call in the render method, we need to check if IconPool is imported.
+	// Currently, we're only able to analyze whether oRm.icon is called in the render method as
+	// there's no reliable way to find if the method icon() is actually a member of RenderManager in other places.
+	analyzeIconCallInRenderMethod(node: ts.Node) {
+		let renderMethodNode: ts.Node | undefined = node;
+
+		// When the render is a plain function
+		if (ts.isMethodDeclaration(node) || ts.isArrowFunction(node) ||
+			ts.isFunctionExpression(node) || ts.isFunctionDeclaration(node)) {
+			renderMethodNode = node;
+		} else if (ts.isObjectLiteralExpression(node)) {
+			// When the render is embed into the renderer object
+			const renderProperty = node.properties.find((prop) => {
+				return ts.isPropertyAssignment(prop) &&
+					ts.isIdentifier(prop.name) &&
+					prop.name.text === "render";
```

Comment:
Side note: In SAPUI5, only 13 of 47 `RenderManager#icon` calls are in the `render` method. The others are in helper methods in the renderer.

Reply:
But is there any way to identify that `oRm.icon(...)` is actually a `RenderManager#icon` call?

Reply:
100% safe? No, not without type info. But I guess it would be quite safe to assume a RenderManager#icon call in the context of a Renderer module (if it has been recognised as such) or within the renderer object in a control (embedded renderer scenario).

Reply:
Now, the detection checks whether `.icon()` call has been made within the `render()` function or within the entire control renderer (if such)